packages/synapse-react-client/src/components/SynapseTable/SynapseTableCell/SynapseTableCell.tsx

```diff
@@ -157,22 +146,29 @@ export function SynapseTableCell(props: SynapseTableCellProps) {
         </p>
       )
     }
-    case ColumnTypeEnum.FILEHANDLEID:
+    case ColumnTypeEnum.FILEHANDLEID: {
+      let associatedObjectId = entity!.id!
+      let associatedObjectType = FileHandleAssociateType.TableEntity
+      if (isFileViewOrDataset(entity) && columnName === 'dataFileHandleId') {
+        associatedObjectId = String(rowId)
+        associatedObjectType = FileHandleAssociateType.FileEntity
+      }
```

Comment:
I think I fixed a bug here where users could not directly download `dataFileHandleId` columns in views (unless they were the file handle uploader)

Reply:
Yes, I don't think accessing the FileHandle.dataFileHandle via the previous FHA (type=TableEntity) would work unless you were the file owner

Reply:
Nice catch